provider/oci/environ.go

```diff
@@ -393,10 +406,17 @@ func (e *Environ) DestroyController(ctx envcontext.ProviderCallContext, controll
 
 	err = e.StopInstances(ctx, ids...)
 	if err != nil {
-		return err
+		providerCommon.HandleCredentialError(err, ctx)
+		return errors.Trace(err)
 	}
 	logger.Debugf("Cleaning up network resources")
-	return e.cleanupNetworksAndSubnets(controllerUUID, "")
+	err = e.cleanupNetworksAndSubnets(controllerUUID, "")
+	if err != nil {
```

Comment:
no need for nil check here, just do
```
providerCommon.HandleCredentialError(err, ctx)
return err
```

as the auth err check func should return false for nil, and the cleanupNetworksAndSubnets shoulf already have applied the trace wrapper

Reply:
I actually put it back in because I wanted to be able to call `errors.Trace(err)`. If you're happy to go without the trace. then this is an easy fix.